src/runtime/mpz.cpp

```diff
@@ -267,6 +267,12 @@ uint64 mpz::mod64() const {
     return (static_cast<uint64>(h.get_unsigned_int()) << 32) + static_cast<uint64>(l.get_unsigned_int());
 }
 
+int mpz::smod8() const {
```

Comment:
Also, the `smod8` name is really suggesting that the result should fit cleanly into an `int8`.

Reply:
`smod8` returning `int` was done because `mod8` returns `unsigned` which seemed consistent. But I can change that as well if you want. I'll look into the invariant.